### ISSUE TITLE
Customize menu action item

### ISSUE THREAD

**Issue by Flam3rboy** [dominicstop/react-native-ios-context-menu#31]
Thank you for creating this amazing package 🚀

I'm currently trying to add custom buttons to the context-menu, because not every emoji reaction should be a separate action menu item. That's why I've made a custom preview render, which renders a small bar with pressables.

<img src="https://user-images.githubusercontent.com/34555296/147880386-d96aa12e-98eb-43aa-94de-68c168d161b6.png" width="300">

However when I try to click it the onPress handles don't get called.
I've tried to use onPressMenuPreview instead, however the page x & y position of the pressed item don't get passed to the event handler, only the generic ``RNIContextMenuView`` tag is passed. 

Is there any way to either customize an action menu item or make the buttons in the preview pressable?

**Comment by dominicstop** [comment]
Hey, unfortunately this is a limitation of the `UIMenu` API 😔

This is actually something I want too, but TLDR the preview doesn't receive any touch events once it's being previewed.

The one in the iMessage app (Reactions), and Safari app (webpage previews), uses private API's.

As soon as apple exposes the public API, I'll add it in haha (maybe if we all file a [radar](https://developer.apple.com/bug-reporting/), they'll finally add it)

Note: the last time I tried making the menu preview interactive was in 2019 though, maybe something has changed internally to allow this

Here is the summary:
* The previews are handled via `UIContextMenuContentPreviewProvider` — it returns a view controller (it contains the view you want to show in the preview)
* But once the context menu interaction begins, the view + view controller no longer receive touch events (they are no longer interactive)
* I even tried setting `view. userInteractionEnabled = true` when the context menu interaction begins but no luck haha

**Comment by nandorojo** [comment]
Ah I wish Apple would just open this awesome API...thanks for looking into it @dominicstop

**Comment by Flam3rboy** [comment]
Ok not much we can do about, but thank you for your effort. Get well soon @dominicstop

**Comment by dominicstop** [comment]
Hey sorry for the late response haha here are results of my testing ✨
```
(lldb) po self.previewController?.next
▿ Optional<UIResponder>
  - some : <UIViewController: 0x7f7a6f90e100>

(lldb) po self.previewController?.next?.next
▿ Optional<UIResponder>
  - some : <UITransitionView: 0x7f7a73016ab0; frame = (0 0; 375 667); alpha = 0; autoresize = W+H; userInteractionEnabled = NO; layer = <CALayer: 0x600002f3dea0>>
```

* `UITransitionView` is internal, not published in the SDK

<br>

```
(lldb) po self.previewController?.next?.next?.next
▿ Optional<UIResponder>
  - some : <UIWindow: 0x7f7a6f90d1e0; frame = (0 0; 375 667); autoresize = W+H; gestureRecognizers = <NSArray: 0x60000265f2d0>; layer = <UIWindowLayer: 0x6000026539c0>>
```

* `UIViewController` (Preview Content) -> `UITransitionView` (Internal) -> `UIWindow` (Main Window)

<br>

```
(lldb) po self.previewController?.presentationController
▿ Optional<UIPresentationController>
  - some : <_UIContextMenuPresentationController: 0x7f7a72365200>

(lldb) po self.previewController?.presentingViewController
▿ Optional<UIViewController>
  - some : <UIViewController: 0x7f7a6f90e100>
  
(lldb) po self.previewController?.presentationController
▿ Optional<UIPresentationController>
  - some : <_UIContextMenuPresentationController: 0x7f7a72365200>

(lldb) po self.previewController?.presentingViewController
▿ Optional<UIViewController>
  - some : <UIViewController: 0x7f7a6f90e100>
```

* This is how you get the  parent view controller of the preview

<br>

```
(lldb) po self.previewController?.presentingViewController?.view
▿ Optional<UIView>
  - some : <RCTRootView: 0x7f7a6ef3ee90; frame = (0 0; 375 667); autoresize = W+H; layer = <CALayer: 0x6000028043a0>>

(lldb) po self.window?.rootViewController?.view
▿ Optional<UIView>
  - some : <RCTRootView: 0x7f7a6ef3ee90; frame = (0 0; 375 667); autoresize = W+H; layer = <CALayer: 0x6000028043a0>>

(lldb) po self.previewController?.presentingViewController?.view === self.window?.rootViewController?.view
true
```

* the preview is being presented using the current window's root view controller

<br>

```
(lldb) po (self.previewController?.next?.next as? UIView)?.gestureRecognizers
nil

(lldb) po (self.previewController?.next?.next as? UIView)
▿ Optional<UIView>
  - some : <UITransitionView: 0x7f7a73016ab0; frame = (0 0; 375 667); alpha = 0; autoresize = W+H; userInteractionEnabled = NO; layer = <CALayer: 0x600002f3dea0>>
```

* `userInteractionEnabled` is disabled, enabling it does nothing...
* **TLDR**: What needs to be done is to become responsible for handling the gestures for the preview controller, but it is not exposed...
* The moment I try to use the private API's, app store connect might reject it (I can do some swizzling, or drop down to Obj-C to use them)
<br>

Debug Layout:
![Screen Shot 2022-01-06 at 6 30 20 PM](https://user-images.githubusercontent.com/18517029/148370732-ffa35091-d696-4ea3-bade-c470b95b1c8d.png)

---
Summary: interactive context menu's are still not possible in iOS 15 😔

**Comment by dominicstop** [comment]
@nandorojo @Flam3rboy its very rough, but here's a proof of concept haha

basically this lets you attach an auxiliary view (that can be interacted with) on top of the edge of the context menu preview

https://user-images.githubusercontent.com/18517029/148428709-cfa2f805-95a2-4ebe-984c-487d68f8c17b.mov

**Comment by Flam3rboy** [comment]
This is awesome, let me know if you add it to the library 🚀

**Comment by dominicstop** [comment]
@Flam3rboy i posted a preview on [twitter](https://twitter.com/godominic/status/1482768021949419522?s=21)

i'm still designing an API around it, but i'm not sure if the app store would allow this though haha

**Comment by fukemy** [comment]
Hi, i done it inside my Swift project, you can see here:

My logic: 


https://user-images.githubusercontent.com/8202437/174036068-76047fb2-83d4-4393-997d-131873c8ffd3.MP4

But i dont know how to do it's in React Native, can u  help? Thanks so much

**Comment by nandorojo** [comment]
Yeah, see `renderAuxiliaryPreview`

**Comment by fukemy** [comment]
ok tks worked

**Comment by tmbiOS** [comment]
@dominicstop  Hi! Thanks for your library! Can you advice how to render Auxiliary Preview using native code -  Swift, maybe alhorithm.
